.github/workflows: refresh

jyhi opened the pull request:
This PR refurbishes workflow files, making them more tidy and sound.

Unsuccessful checks are disabled, and will be enabled one by one after the corresponding problems are solved in the source code. See #782.

Fixes:

- missing conditions triggering workflows
- inconsistent copy-pastes across workflows
- comments by adding clear names to jobs and tasks
- caching that did not work across parallel jobs
- builds and releases on outdated Node.js and Python versions
- tools used for testing
- missing coverages by adding more build and test checks
- organization of workflows by using a per-package approach

This PR depends on #784.

jyhi committed:
.github/workflows: refresh

Refurbish workflow files, making them more tidy and sound.

Unsuccessful checks are disabled, and will be enabled one by one
after the corresponding problems are solved in the source code.
See #782.

Fixes:

- missing conditions triggering workflows
- inconsistent copy-pastes across workflows
- comments by adding clear names to jobs and tasks
- caching that did not work across parallel jobs
- builds and releases on outdated Node.js and Python versions
- tools used for testing
- missing coverages by adding more build and test checks
- organization of workflows by using a per-package approach

Thoralf-M commented on .github/actions/setup-rust/action.yml:
```diff
@@ -0,0 +1,91 @@
+    description: Target triple of the toolchain.
+    type: string
+    required: false
+    default:
```
Is this needed if it's just empty?

jyhi replied:
This is just an explicit empty default, actually redundant.

Thoralf-M commented on .github/workflows/python-bindings-publish.yml:
```diff
@@ -0,0 +1,57 @@
+      - name: Upload Wheels
+        uses: actions/upload-artifact@v2
+        with:
+          name: ${{ matrix.identifier }}-iota-wallet-py${{ matrix.python }}-wheel
```
```suggestion
          name: ${{ matrix.identifier }}-iota-client-py${{ matrix.python }}-wheel
```

jyhi replied:
Oh no.

Thoralf-M commented on .github/workflows/examine-core.yml:
```diff
@@ -0,0 +1,153 @@
+      - 'Cargo.toml'
+      - 'Cargo.lock'
+  schedule:
+    - cron: '0 1 * * *'
```
With that it will run automatically every day, additionally to the pull requests to the dev branch?

jyhi replied:
Additionally every job has an `if` statement constraining when they'll be run. For example, none but `audit` will be run every day because all other jobs are marked as `if: ${{ ! github.event.schedule }}`.

This is actually a trick to merge the auditing workflow with the crate-wise workflows. `audit.yml` used to be a separate workflow triggered every day. Now, all crate-wise workflows are triggered every day, but when the workflows are run because of the `schedule` event, only the `audit` job will be run. This reduces redundancy.

jyhi committed:
.github/workflows: improve paths

- Trigger all crate workflows when the Rust core is changed
- Misc. refinements

Thoralf-M commented on .github/workflows/python-bindings-publish.yml:
```diff
@@ -5,11 +5,13 @@ on:
-      - src/**
-      - Cargo.lock
-      - Cargo.toml
+      - '.github/workflows/examine-wasm-bindings.yml'
```
This should stay python, no?

jyhi replied:
:(

jyhi committed:
.github/workflows: fix wrong path

jyhi committed:
.github/workflows: fix wrong python wheel name

rajivshah3 commented on .github/workflows/covector.yml:
```diff
@@ -0,0 +1,156 @@
+    paths-ignore:
+      - 'specs/**'
+      - 'documentation/**'
+  pull_request:
```
I might be wrong but does this need to run on PRs? The `status` job just seems to [provide outputs](https://github.com/jbolda/covector/tree/covector-v0/packages/action#outputs) for `version-or-publish` which should only run on push events

jyhi replied:
Hmm.

rajivshah3 commented on .github/workflows/python-bindings-publish.yml:
```diff
@@ -0,0 +1,59 @@
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest, macos-latest, windows-latest]
```
```suggestion
        os: [ubuntu-18.04, macos-latest, windows-latest]
```

I think we build on Ubuntu 18.04 to avoid issues with glibc (#588)

jyhi replied:
Fair enough.

rajivshah3 commented on .github/workflows/covector.yml:
```diff
@@ -0,0 +1,156 @@
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest, macos-latest, windows-latest]
```
```suggestion
        os: [ubuntu-18.04, macos-latest, windows-latest]
```

I think we build on Ubuntu 18.04 to avoid issues with glibc (#576)

jyhi committed:
.github/workflows: don't run covector on pr

jyhi committed:
.github/workflows: use ubuntu-18.04 for releasing

jyhi committed:
.github/workflows: disable wasm lint
